### PR TITLE
Fix compiling with non-MPI PETSc

### PR DESCRIPTION
**Pull request by masterleinad**

### CODE REVIEW

**Comment by drwells** [comment]
Thanks for fixing this!

File: source/base/mpi.cc [hunk 1]
```diff
@@ -324,12 +307,12 @@ namespace Utilities
                           "in a program since it initializes the MPI system."));
 
 
-
+      int ierr;
```
Comment (drwells) [inline source/base/mpi.cc]: Is there a reason that you moved this here?

Comment (davydden) [inline source/base/mpi.cc]: i like better the previous location. I think @drwells unified those a while ago, let's keep it homogeneous within the library.

Comment (masterleinad) [inline source/base/mpi.cc]: We are reusing `ierr` for `PETScInitialize` in line 348 and hence this was not compiling using a non-MPI PETSc.
But this way `ierr` is unused if we use neither `MPI` nor `PETSc`. This is now fixed as well.

**Comment by bangerth** [comment]
I see why @masterleinad moved the declaration of the variable. Once could have duplicated and only use it in its own scope, but it's not worth going to lengths with this.

**Comment by drwells** [comment]
Ah, okay: I couldn't find the other usage.

**Pull request merged**